builtin/logical/pki/backend.go

```diff
@@ -252,19 +258,52 @@ func (b *backend) metricsWrap(callType string, roleMode int, ofunc roleOperation
 
 // initialize is used to perform a possible PKI storage migration if needed
 func (b *backend) initialize(ctx context.Context, req *logical.InitializationRequest) error {
-	logger := b.Logger().Named("initialize")
-
 	// Early exit if not a primary cluster or performance secondary with a local mount.
 	if b.System().ReplicationState().HasState(consts.ReplicationDRSecondary|consts.ReplicationPerformanceStandby) ||
 		(!b.System().LocalMount() && b.System().ReplicationState().HasState(consts.ReplicationPerformanceSecondary)) {
-		logger.Debug("skipping pki migration as we are not on primary or secondary with a local mount")
+		b.Logger().Debug("skipping pki migration as we are not on primary or secondary with a local mount")
 		return nil
 	}
 
-	if err := migrateStorage(ctx, req, logger); err != nil {
-		logger.Error("Error during migration of PKI mount: " + err.Error())
+	if err := migrateStorage(ctx, req, b.Logger()); err != nil {
+		b.Logger().Error("Error during migration of PKI mount: " + err.Error())
 		return err
 	}
 
+	b.updatePkiStorageVersion(ctx)
+
 	return nil
 }
+
+func (b *backend) useLegacyBundleCaStorage() bool {
+	version := b.pkiStorageVersion.Load()
+	return version == nil || version == 0
+}
+
+func (b *backend) updatePkiStorageVersion(ctx context.Context) {
+	info, err := getMigrationInfo(ctx, b.storage)
+	if err != nil {
+		b.Logger().Error(fmt.Sprintf("Failed loading PKI migration status, staying in legacy mode: %v", err))
```

Comment:
And related to the above, wondering if that means the migration status will be incorrect here? Namely, the `if !info.isRequired {...}` check later might need to be moved up?

Reply:
If you are referring to the empty/non-existant ca-bundle then that won't be a problem. Please let me know if I have misunderstood your concern.

Reply:
Yeah, it was about the above comment, but I see that's wrong, feel free to ignore this one :-)